include/gridtools/common/hugepage_alloc.hpp

```diff
@@ -11,25 +11,178 @@
 #pragma once
 
 #include <atomic>
+#include <cassert>
 #include <cstdlib>
+#include <cstring>
 #include <new>
+#include <stdexcept>
+#if __cpp_lib_int_pow2 >= 202002L
+#include <bit>
+#endif
+#ifdef __linux__
+#include <cstdio>
+
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#endif
 
 namespace gridtools {
+    namespace hugepage_alloc_impl_ {
+        inline std::size_t ilog2(std::size_t i) {
+#if __cpp_lib_int_pow2 >= 202002L
```

Comment:
I would just use the hand rolled version with the `while` loop, to avoid the `#ifdef`.

Reply:
Done.